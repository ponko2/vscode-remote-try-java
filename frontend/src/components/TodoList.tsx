import clsx from "clsx";
import type { components } from "../api/schema";
import { TodoItem } from "./TodoItem";

type Props = {
  todos: components["schemas"]["TodoResponse"][];
  deleteTodo: (id: string) => void;
  toggleTodo: (data: components["schemas"]["TodoResponse"]) => void;
  updateTodo: (data: components["schemas"]["TodoResponse"]) => void;
};

export function TodoList({ todos, deleteTodo, toggleTodo, updateTodo }: Props) {
  return (
    <section>
      <ul className={clsx("divide-y", "divide-neutral-200")}>
        {todos.map((todo) => (
          <TodoItem
            deleteTodo={deleteTodo}
            key={todo.id}
            todo={todo}
            toggleTodo={toggleTodo}
            updateTodo={updateTodo}
          />
        ))}
      </ul>
    </section>
  );
}
