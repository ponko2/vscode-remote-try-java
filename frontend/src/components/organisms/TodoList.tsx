import { TodoItem } from "../../components/molecules/TodoItem";
import { TodoPatchRequestModel, TodoResponseModel } from "../../models/todos";
import styles from "./TodoList.module.css";

type Props = {
  todos: TodoResponseModel[];
  deleteTodo: (id: string) => void;
  editTodo: (data: TodoPatchRequestModel) => void;
  toggleTodo: (data: TodoPatchRequestModel) => void;
};

export function TodoList({
  todos,
  deleteTodo,
  editTodo,
  toggleTodo,
}: Props): JSX.Element {
  return (
    <section className={styles.todoList}>
      <ul>
        {todos.map((todo) => (
          <TodoItem
            key={todo.id}
            todo={todo}
            editTodo={editTodo}
            toggleTodo={toggleTodo}
            deleteTodo={deleteTodo}
          />
        ))}
      </ul>
    </section>
  );
}
