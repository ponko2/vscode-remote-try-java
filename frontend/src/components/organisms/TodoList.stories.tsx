import { action } from "@storybook/addon-actions";
import type { Meta, StoryFn } from "@storybook/react";
import { TodoList } from "./TodoList";

export default {
  component: TodoList,
} as Meta<typeof TodoList>;

const Template: StoryFn<typeof TodoList> = (args) => (
  <TodoList
    {...args}
    deleteTodo={action("deleteTodo")}
    editTodo={action("editTodo")}
    toggleTodo={action("toggleTodo")}
  />
);

export const Basic = Template.bind({});

Basic.args = {
  todos: [
    {
      id: "01FYH5XVSNVSXTSGB8KB858REF",
      title: "foo",
      completed: false,
    },
    {
      id: "01G46BYCGQ1SGVGFMEXZ0DKZAY",
      title: "bar",
      completed: true,
    },
    {
      id: "01G46BZM28F68BCY7EP016G1EZ",
      title: "baz",
      completed: false,
    },
  ],
};
