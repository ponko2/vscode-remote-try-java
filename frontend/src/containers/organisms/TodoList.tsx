import { TodoList as Component } from "../../components/organisms/TodoList";
import {
  useDeleteTodo,
  useEditTodo,
  useTodos,
  useToggleTodo,
} from "../../hooks/todos";
import { TodoIndexResponseModel, TodoResponseModel } from "../../models/todos";

type Props = {
  select?: (data: TodoIndexResponseModel) => TodoResponseModel[];
};

export function TodoList({ select }: Props): JSX.Element {
  const { data: todos = [] } = useTodos(select);
  const editTodo = useEditTodo();
  const toggleTodo = useToggleTodo();
  const deleteTodo = useDeleteTodo();
  return (
    <Component
      todos={todos}
      editTodo={editTodo.mutate}
      toggleTodo={toggleTodo.mutate}
      deleteTodo={deleteTodo.mutate}
    />
  );
}
