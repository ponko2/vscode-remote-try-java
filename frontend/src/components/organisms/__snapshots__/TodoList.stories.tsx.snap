// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/TodoList Basic smoke-test 1`] = `
<section class="_todoList_1u114_1">
  <ul>
    <li class="_todoItem_17lsu_1">
      <div class="_view_17lsu_69">
        <input class="_toggle_17lsu_34"
               type="checkbox"
        >
        <label>
          foo
        </label>
        <button class="_button_z78gq_1 _destroy_17lsu_79">
        </button>
      </div>
    </li>
    <li class="_todoItem_17lsu_1">
      <div class="_view_17lsu_69">
        <input class="_toggle_17lsu_34"
               type="checkbox"
               checked
        >
        <label>
          bar
        </label>
        <button class="_button_z78gq_1 _destroy_17lsu_79">
        </button>
      </div>
    </li>
    <li class="_todoItem_17lsu_1">
      <div class="_view_17lsu_69">
        <input class="_toggle_17lsu_34"
               type="checkbox"
        >
        <label>
          baz
        </label>
        <button class="_button_z78gq_1 _destroy_17lsu_79">
        </button>
      </div>
    </li>
  </ul>
</section>
`;
